<header class="banner"><div class="banner">
  <div class="logos">
    <div><img title="Julia logo" src="../figures/julia-logo-color.svg" alt="Julia animated logo" width=35 /></div>
    <div>◌</div>
    <div><img title="Powered by MathJax" src="https://www.mathjax.org/badge/mj-logo.svg"
              alt="Powered by MathJax" width=65 />
    </div>
  </div>  <!-- logos -->
  <div class="cite_textbook">me215 breakout session, 29 march 2021 <span class="authors"></span></div>
</div></header>

<footer class="navbar navbar-default navbar-fixed-bottom mcs-psy-nav" >
  <div class="container-fluid text-right h4">
    <em>ψᵟ</em>
  </div>
</footer>



### Given 

A 2 m³ rigid insulated tank, initially evacuated, is connected through a valve to a supply line that carries saturated vapor water at 1 MPa. Now the valve is opened, and steam is allowed to flow slowly into the tank until the pressure in the tank rises to 1 MPa. 

<img src="../figures/20210329⋮breakout⋮supply_line.pdf" alt="supply line" width="50%" />


### Find
<ol>
  
1. the mass of the steam that has entered, and
1. the final temperature of the steam in the tank.


### Get Organized

First, we observe that this is NOT a steady state process. Energy and/or mass of the system are changing in time. Remember, we must account for our system spatially and temporally. It's not difficult if you stay organized. We are given enough information to resolve two of the three states we need (_initial, final, supply_). Let's go through all the information we need to find in order to answer the questions.  
<ol type="A" >
    
1. **Mass entered   $m_{in}$**   
   Not so bad. Apply continuity (aka conservation of mass)   
  
  \begin{gather}
    m_{in} &-& m_{out} &=& \Delta m_{sys}   \\[6pt]
    m_{in} && &=& m_{\it final}   \label{eq:continuity} \\
  \end{gather}  

  All we need is $m_{\it final}.$ No problem: We were given $V,$ and we know $V = m_{\it final}v_{\it final}.$ So, we resolve the final state to get $v_1$, then we can calculate $m_{in}=m_{\it final}$. 
  
  Now, about that final state...
  <br><br>
  
1. **Final temperature $T_{\it final}$**   
   Apply the energy equation, considering what flows we have crossing the boundary and the energy theu advect with them. Of course, we also consider what forms of energy (internal, flow, kinetic, potential) are important to the system itself (internal, kinetic, potential).   

   \begin{gather}
    E_{in} &-& E_{out} &=& \Delta E_{sys}                                             \\[6pt]
    E_{mass,in} && &=& U_{final}                                               \notag \\[6pt]
    m_{in}\big(u_{in} + P_{in}v_{in}\big) &&  &=& m_{\it final}u_{\it final}   \notag \\[6pt]
    m_{in}h_{in} &&  &=& m_{\it final}u_{\it final}  \label{eq:firstlaw}              \\[6pt]
   \end{gather} 
  
   From eqn \ref{eq:continuity}, 
  
   \begin{gather*}
    m_{in} = m_{\it final}.  
   \end{gather*} 
   
   Thus, eqn \ref{eq:firstlaw} becomes
   
   \begin{align}
    m_{in}h_{in} &= m_{\it in}u_{\it final}  \notag \\[6pt]
    h_{in} &= u_{\it final}  \label{eq:energydone}  \\[6pt]
   \end{align} 

   We're home free. We need only resolve state 2 to get $T_{\it final}$ and $v_{\it final}$, which we subsequently use to calculate the mass that entered our system.

### Attack
Dive into the property table, and take eqn \ref{eq:energydone}

\begin{equation}
  u_{\it final} = h_{in} = h_{g@1 \rm MPa} 
\end{equation}

Interpolate $T_{\it final}, v_{\it final}:$

\begin{equation}
   \frac{u_{\it final} - u_\text{low}}{u_\text{high} - u_\text{low}} 
 = u_\text{fraction}
 = \frac{T_{\it final} - T_\text{low}}{T_\text{high} - T_\text{low}} 
 = \frac{v_{\it final} - v_\text{low}}{v_\text{high} - v_\text{low}} 
\end{equation}




In [1]:
using Unitful  # Julia units package

# given
V = 2u"m^3"

# tables
hg_1MPa = 2778.1u"kJ/kg"

T_low  = 280u"°C" |> u"K"  # Unitful does not like relative temperature calculations
T_high = 320u"°C" |> u"K"

u_final = hg_1MPa
u_1MPa_280°C = 2760.2u"kJ/kg";  u_low  = u_1MPa_280°C
u_1MPa_320°C = 2826.1u"kJ/kg";  u_high = u_1MPa_320°C

v_1MPa_280°C = 0.2480u"m^3/kg"; v_low  = v_1MPa_280°C
v_1MPa_320°C = 0.2678u"m^3/kg"; v_high = v_1MPa_280°C


# ATTACK
u_fraction = (u_final - u_low) / (u_high - u_low)

T_final = T_low + u_fraction * (T_high - T_low) |> u"°C"

v_final = v_low + u_fraction * (v_high - v_low)

m_in = V / v_final 

@show m_in T_final;

m_in = 8.064516129032258 kg
T_final = 290.8649468892262 °C


<div class="mcs-bs-callout mcs-bs-callout-success">

#### Found

\begin{align*}
m_{in} &= \pu{8.065 kg}  \\[6pt]
T_{\it final} &= 290.865\,\rm °C
\end{align*}
    
</div>

<div class="mcs-bs-callout mcs-bs-callout-tiger extra">

#### So why did the temperature increase?

</div>

<footer style="background-color: #461d7c; color:rgba(253,208,35,0.35); line-height:2em; padding-left:1em; padding-right:0.5em; bottom:0; float:bottom; text-align:right;"><i>ψᵟ</i></footer>